Add page for other helpers

Gaurav0 opened the pull request:
This is for issue #1249 

Please bear with me as this is the first time I'm adding a page to the guides. This page is not intended to be complete, but a start, so please excuse if any helpers are missing.

This page is basically going to be a place to go ahead and add miscellaneous helpers until a better place is found for them.

locks commented:
Can you add a newline after each period, please? You missed a couple, it makes it easier to give specific feedback.

locks commented on source/localizable/templates/other-helpers.md:
```diff
@@ -0,0 +1,60 @@
+
+### Concat
+
+The [`{{concat}}`][1] helper makes it easy to dynamically send two parameters
```
beware, concat can receive N number of arguments, not just two.

Gaurav0 commented:
@locks I have updated this in accordance with our conversation on slack. I have moved it to the top of the templates page and made it fit better. I have removed the `hash` helper as I think it belongs with the page on contextual components.

locks commented on source/localizable/templates/some-basic-helpers.md:
```diff
@@ -0,0 +1,42 @@
+## Some Basic Built in Helpers
+
+In the last section you learned how to write a helper.
```
I just read the previous section, and it feels weird to both have it before this section, and then have this section, and then go back to writing helpers.
Can you move the "Helpers" section from that file to the end of this one? Connect both of them by saying something like "Besides using the built-in helpers, you can also write your own"

locks replied:
ignore the previous comment for now. address the other ones and we'll merge :]

locks commented on source/localizable/templates/some-basic-helpers.md:
```diff
@@ -0,0 +1,42 @@
+## Some Basic Built in Helpers
```
"Built-in Helpers" would suffice

locks replied:
@acorncom can you confirm that "built-in" is properly hyphenated?

locks commented on source/localizable/templates/some-basic-helpers.md:
```diff
@@ -0,0 +1,42 @@
+## Some Basic Built in Helpers
+
+In the last section you learned how to write a helper.
+A helper, unlike a component, is usually a simple function that can be
```
we haven't introduced components yet, the comparison seems unnecessary

locks commented on data/pages.yml:
```diff
@@ -84,6 +84,8 @@
   pages:
     - title: "Handlebars Basics"
       url: "handlebars-basics"
+    - title: "Some basic Helpers"
```
can you rename to "Built-in Helpers"?

locks commented:
Thanks for your contribution!

@homu r+

homu commented:
:pushpin: Commit 97c02aa has been approved by `locks`

<!-- @homu r=locks 97c02aab8946dbdc97cb056a59be8cacc05a97a2 -->

homu commented:
:hourglass: Testing commit 97c02aa with merge c1cedeb...

homu commented:
:sunny: Test successful - [status](https://travis-ci.org/emberjs/guides/builds/131957702)
